shared/src/main/scala/kinesis4cats/producer/Producer.scala

```diff
@@ -182,8 +182,7 @@ abstract class Producer[F[_], PutReq, PutRes] private[kinesis4cats] (
       )
       finalRes <- retryingOnFailuresAndAllErrors(
         config.retryPolicy,
-        (x: Producer.Result[PutRes]) =>
-          F.pure(x.isSuccessful || (x.isPartiallySuccessful && !x.hasFailed)),
+        (x: Producer.Result[PutRes]) => F.pure(!x.hasFailed),
```

Comment:
Does this simplification seem reasonable? Thinking here is that the only time we need to retry is if there are any failed records. There is no point retrying invalid records, and we obviously don't need to try successful records.

Reply:
This does seem reasonable to me.